public_html/wp-content/plugins/wcpt/wcpt-event/class-event-admin.php

```diff
@@ -694,6 +697,59 @@ public function print_admin_notices() {
 
 	}
 
+	/**
+	 * Prevent users from creating new WordCamp and Meetup posts on dashboard. In most of the cases, all posts should be created
+	 * thru the public application forms in order to get all needed information and to initiate the vetting process correctly.
+	 *
+	 * Expectation is made for users with administrator and deputy roles, as they need to create events manually from time to time.
+	 *
+	 * Used wp_insert_post_empty_content hook is run fo creation and updates, which is why post ID needs to be checked. The hook
+	 * short circuits creation of new post when truthy value is returned.
+	 *
+	 * @param  boolean $maybe_empty Whether the post should be considered "empty".
+	 * @param  array   $postarr      Array of post data.
+	 *
+	 * @return boolean              Whether the post should be considered "empty".
+	 */
+	public function maybe_prevent_creation_of_new_post( $maybe_empty, $postarr ) {
+		$post_type = $postarr['post_type'];
+
+		// Apply only for WordCamp and Meetup post types.
+		if ( $this->get_event_type() !== $post_type ) {
+			return $maybe_empty;
+		}
+
+		// The action hooked into is used also when updating posts, which all users should be able to do based on their caps.
+		if ( ! empty( $postarr['ID'] ) ) {
+			return $maybe_empty;
+		}
+
+		// Doing the checks only on dashboard ensures that other use cases like application forms do still work.
+		if ( ! is_admin() ) {
```

Comment:
This will also return TRUE for ajax requests, do we want to also consider this?

Reply:
Can't think of places where new event posts would be created via ajax request. And if someone manages to do that, it's not that a big deal - deputies do catch these manual entries.